src/appComponents/Routes/index.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Switch, Route } from 'react-router'
+import Home from '../../pages/Home';
+import Favorites from '../../pages/Favorites';
+import View from '../../pages/View';
+import NotFound from '../../pages/NotFound';
+
+const Routes = () => (
+  <Switch>
+    <Route exact path="/" component={Home}/>
+    <Route path="/favorites" component={Favorites}/>
+    <Route path="/view/:gif" component={View}/>
+    <Route path="*" component={NotFound}/>
```

Comment:
You can define a `<Route component={NotFound} />`(without a path) at the end and the Switch with default to it if there is no other match before